src/client/Microsoft.Identity.Client/WsTrust/WsTrustWebRequestManager.cs

```diff
@@ -101,8 +101,14 @@ public async Task<MexDocument> GetMexDocumentAsync(string federationMetadataUrl,
             }
             catch (System.Xml.XmlException ex)
             {
+                string message = string.Format(
+                        CultureInfo.CurrentCulture,
+                        MsalErrorMessage.FederatedServiceParseErrorTemplate,
+                        wsTrustEndpoint.Uri,
+                        resp.Body);
+
                 throw new MsalClientException(
-                    MsalError.ParsingWsTrustResponseFailed, MsalError.ParsingWsTrustResponseFailed, ex);
+                    MsalError.ParsingWsTrustResponseFailed, message, ex);
```

Comment:
this is awesome. Thanks @jabbera for the contribution. Do you know if there would be PII data ever in the response body?
LGTM.

Reply:
Well, it's Microsoft's endpoint so out of the box maybe you can tell me since I haven't been able to get my firewall team to sort this yet:-)

As for what our firewall puts in, it's internal addresses and usernames, but I'd guess that's specific to each company.

Reply:
Renamed the error template to match the error being raised. (Seems to be your convention) and added a test. Happy to squash if you want.

Reply:
Thanks! We can squash the commits when we merge. Appreciate the text coverage.